addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/internal/SmappeeService.java

```diff
@@ -0,0 +1,461 @@
+                    consumption.solar = consumption.solar * 12;
+                }
+
+                logger.debug("smappee'{}' read", this.serviceLocationId);
+
+                return readings;
+
+            } catch (Exception se) {
+                logger.error("failed to read smappee '{}'", se.getMessage());
+            }
+        } while (currentTry++ < this.retry);
+
+        return null;
+    }
+
+    public SmappeeServiceLocationInfo getServiceLocationInfo() {
+        int currentTry = 0;
+        do {
+            try {
+                // sample API method to call :
+                // https://app1pub.smappee.net/dev/v1/servicelocation/123/info
+
+                String responseReadings = GetData("/dev/v1/servicelocation/" + this.serviceLocationId + "/info");
+
+                SmappeeServiceLocationInfo readings = gson.fromJson(responseReadings, SmappeeServiceLocationInfo.class);
+
+                logger.debug("servicelocationinfo '{}' read", this.serviceLocationId);
+
+                return readings;
+
+            } catch (Exception se) {
+                logger.error("failed to read servicelocationinfo '{}'", se.getMessage());
+            }
+        } while (currentTry++ < this.retry);
+
+        return null;
+    }
+
+    /**
+     * Gets the smappee energy consumption readings for a specific appliance.
+     *
+     * @return the device readings
+     */
+    public SmappeeApplianceEvent getLatestApplianceReading(String applianceId) {
+        int currentTry = 0;
+        do {
+            try {
+                DateTime nowUtc = DateTime.now(DateTimeZone.UTC).minusMinutes(1);
+                DateTime nowUtcMinus1Year = DateTime.now(DateTimeZone.UTC).minusYears(1);
+
+                String nowUtcMillis = String.valueOf(nowUtc.getMillis());
+                String nowUtcMinus1YearMillis = String.valueOf(nowUtcMinus1Year.getMillis());
+
+                // sample API method to call :
+                // https://app1pub.smappee.net/dev/v1/servicelocation/123123/events?
+                // applianceId=1&applianceId=2&maxNumber=100&from=1388534400000&to=1391212800000
+
+                String responseReadings = GetData(
+                        "/dev/v1/servicelocation/" + this.serviceLocationId + "/events?applianceId=" + applianceId
+                                + "&maxNumber=1&from=" + nowUtcMinus1YearMillis + "&to=" + nowUtcMillis);
+
+                SmappeeApplianceEvent[] readings = gson.fromJson(responseReadings, SmappeeApplianceEvent[].class);
+
+                logger.debug("smappee '{}' - appliance '{}' read", this.serviceLocationId, applianceId);
+
+                if (readings != null && readings.length == 1) {
+                    return readings[0];
+                }
+                return null;
+
+            } catch (Exception se) {
+                logger.error("failed to read smappee '{}' - appliance '{}' : {}", this.serviceLocationId, applianceId,
```

Comment:
Same comment about catching Exception, logging to error and exception as last argument.

Reply:
ok, refactored